src/gui/sdlmain.cpp

```diff
@@ -3057,20 +3059,27 @@ static void launchcaptures(std::string const& edit) {
 	}
 	Cross::CreatePlatformConfigDir(path);
 	path += file;
-	Cross::CreateDir(path);
-	struct stat cstat;
-	if(stat(path.c_str(),&cstat) || (cstat.st_mode & S_IFDIR) == 0) {
-		printf("%s doesn't exists or isn't a directory.\n",path.c_str());
-		exit(1);
+
+	if (create_dir(path.c_str(), 0700) != 0) {
+		bool dir_exists = false;
+		if (errno == EEXIST) {
+			struct stat cstat;
+			if ((stat(path.c_str(), &cstat) == 0) &&
+			    (cstat.st_mode & S_IFDIR))
+				dir_exists = true;
+		}
+		if (!dir_exists) {
+			char desc[STRERR_LEN];
+			safe_strerror(desc, errno);
```

Comment:
It would be nice to avoid the `char desc[STRERR_LEN];` whenever we want to use strerror. 

If we're wrapping it, can the description be included (static char[])?

An interface like:
- `std::string safe_strerror(desc)` or
- `const char * safe_strerror(desc)` 

Would be cleaner and less to remember.

Reply:
Ehh, I wanted to avoid that so we could transition to C11 `strerror_s` in the future, but it's even more complicated :/

Implemented `std::string safe_strerror(errno)` but I have no idea if it works outside of Linux, we need testing on Windows, macOS and perhaps a BSD...

Reply:
This also looks great! 

I guess Google's test framework is a small nightmare to get working on macOS and Windows.

 I can do some testing on Windows. Maybe some macOS community members can lend their help.

Reply:
On macOS: I simply haven't tried yet - it seems like it is missing from brew. On Windows it's not hard to get going via VS, but a whole new project needs to be added to solution - I have this work started on an older branch.

Additionally, it seems like gtest maintainers changed their mind and now they recommend bundling source code of test framework with the repository :/ maybe we'll use a Git module for this in the future, if I'll be able to make it work with VS.

Testing is simple: 
```
rm -rf ~/.config/dosbox/capture
touch ~/.config/dosbox/capture
./src/dosbox -opencaptures echo
```
Correct output should say something like this on stderr: `Can't access capture dir '/home/dreamer_/.config/dosbox/capture': File exists`